pages/docs/recipes/adding-a-left-panel.mdx

```diff
@@ -0,0 +1,320 @@
+import Image from "next/image";
+import { Steps } from "nextra-theme-docs";
+
+# Adding a left panel
+
+The [left panel](https://zeroheight.com/23a080e38/p/796c6a-left-panel) provides navigation within an App in O3. It is closely based on Carbon's [UI shell left panel](https://carbondesignsystem.com/components/UI-shell-left-panel/usage/), and is positioned below the header, fixed to the left edge of the page. It is possible to add a left panel to any app in O3 if you need to leverage its navigation capabilities.
+
+This guide will walk you through the process of building a left panel for an example frontend module that handles the management of beds in wards at a medical facility.
+
+## Example: Adding a left panel to the Bed Management app
+
+The Bed Management app requires a left panel showing two links:
+
+- A link named `Summary` which is where you first land after visiting the `/bed-management/summary` route.
+- A link named `Ward allocation` which maps to the `/bed-management/administration` route.
+
+Below is a screenshot of how the landing page looks like:
+
+<br />
+<Image
+  src="/left-panel-home.png"
+  alt="Screenshot of the bed management app landing page showing the left panel"
+  width={1000}
+  height={1000}
+/>
+<br />
+
+To achieve this, we're going to follow the steps below:
+
+<Steps>
+### Step 1: Set up the left panel
+
+We'll start by setting up the left panel. To do so, we'll create a directory named `left-panel` and add a component that renders the left panel.
+
+```tsx copy
+// src/left-panel/left-panel.component.tsx
+import React, { useEffect } from "react";
+import {
+  attach,
+  detach,
+  ExtensionSlot,
+  isDesktop,
+  useLayoutType,
+} from "@openmrs/esm-framework";
+import { SideNav } from "@carbon/react";
+import styles from "./left-panel.scss";
+
+const LeftPanel: React.FC = () => {
+  const layout = useLayoutType();
+
+  useEffect(() => {
+    attach("nav-menu-slot", "bed-management-left-panel");
+    return () => detach("nav-menu-slot", "bed-management-left-panel");
+  }, []);
+
+  return (
+    isDesktop(layout) && (
```

Comment:
Should the example include how we handle this on tablet? Or is this built in?

Reply:
It's built-in.